### PR TITLE
Feature: getting tasks from API

### PR DESCRIPTION
**Pull request by PeytonBoggs**
This branch creates a baseline for the application's connection to the API as well as outputs the contents of the database in HTML. A state called taskList and an interface called Task are created to hold the tasks that are fetched from the API. Then, some basic HTML displays the tasks in the App.

### CODE REVIEW

**Comment by PeytonBoggs** [comment]
Screenshot 2023-07-06 at 1 38 25 PM

File: src/App.tsx [hunk 1]
```diff
@@ -1,6 +1,34 @@
+import { useState, useEffect } from 'react';
+import { Task } from './Task-Interface';
+
 function App() {
+    const [taskList, setTaskList] = useState<Task[]>([]);
+
+    useEffect(() => {
+        fetch("http://localhost:8080/tasks?id=&title=&complete=")
```
Comment (matt-j-deasy) [inline src/App.tsx]: I prefer to make this URL a constant so that its easier to change later. So define this URL somewhere else (like as a const in this file) and use that here.

File: src/App.tsx [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+        .then(response => response.json())
+        .then(data => {
+            setTaskList(data);
```
Comment (matt-j-deasy) [inline src/App.tsx]: You can shorthand this with
```.then(setTaskList)``` 
This infers that you want to pass the result of the first call as a parameter of the second

File: src/App.tsx [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+        });
+    }, []);
+
+    function handleComplete(complete: boolean): string {
```
Comment (matt-j-deasy) [inline src/App.tsx]: Arguable this function could be moved to another file, but this is fine for now

File: src/App.tsx [hunk 4]
```diff
@@ -1,6 +1,34 @@
+import { useState, useEffect } from 'react';
+import { Task } from './Task-Interface';
+
 function App() {
+    const [taskList, setTaskList] = useState<Task[]>([]);
```
Comment (matt-j-deasy) [inline src/App.tsx]: Your App component should not be responsible for rendering items and handling data. Lets move this to another component and have the app render that

**Review by matt-j-deasy — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -1,6 +1,12 @@
+import Title from './Title'
+import TaskList from './TaskList';
+
 function App() {
     return (
-        <h1>Todo List</h1>
+        <div>
+            <Title />
```
Comment (matt-j-deasy) [inline src/App.tsx]: This is usually called AppHeader not Title

File: src/TaskList.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react';
+import { Task } from './Task-Interface';
+
+export default function TaskList() {
+    const [taskList, setTaskList] = useState<Task[]>([]);
+
+    const getTasksURL: string = "http://localhost:8080/tasks?id=&title=&complete="
```
Comment (WesleyChenRTS2) [inline src/TaskList.tsx]: I'd like to see the domain & port come from an .env file, but that can be updated in another pr.

**Review by WesleyChenRTS2 — APPROVED** [review]

File: src/TaskList.tsx (hunk 6 above)
Comment (matt-j-deasy) [inline src/TaskList.tsx]: If you declare this variable inside the Functional Component (aka inside the function TaskList) then it gets created when this component is created the first time. Since this url wont change at all during the lifecycle of the app, you can declare it outside the function. I usually do it at the top of the file and for constants like this, I use all caps. This could be:
GET_TASKS_URL

File: src/Title.tsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export default function Title() {
```
Comment (matt-j-deasy) [inline src/Title.tsx]: Yeah call this AppHeader

**Review by matt-j-deasy — CHANGES_REQUESTED** [review]

File: src/TaskList.tsx (hunk 6 above)
Comment (PeytonBoggs) [reply src/TaskList.tsx]: Eventually, when the search bar is implemented, I think I will want this URL to change (adding id, title, and complete values after the "="), but I will declare it outside of the function for now.
